packages/cdktf-cli/bin/cmds/ui/terraform-context.tsx

```diff
@@ -330,36 +397,66 @@ export const useTerraform = ({ targetDir, synthCommand }: UseTerraformInput) =>
     return state
   }
 
-  const deploy = (plan?: TerraformPlan) => {
+  const deploy = () => {
     React.useEffect(() => {
       const invoke = async () => {
-        if (plan) {
-          await execTerraformApply(plan)
-        } else {
-          throw new Error("expected plan to be present but was undefined")
-        }
+        await execTerraformSynth()
+      }
+      invoke()
+    }, [])
+
+    React.useEffect(() => {
+      const invoke = async () => {
+        await execTerraformInit()
+        await execTerraformPlan(false)
+      }
+      if (state.status === Status.SYNTHESIZED) invoke();
```

Comment:
Am I reading correctly that those won't be ran if a plan has already been created?

Reply:
Yes, whenever any of `[state.status, terraform]` changes, this will be invoked. Therefore this additional guard clause.